src/custom/utils/metadata.ts

```diff
@@ -49,8 +47,6 @@ export function generateAppDataDoc(metadata: MetadataDoc = {}): AppDataDoc {
 }
 
 export async function uploadMetadataDocToIpfs(appDataDoc: AppDataDoc): Promise<string> {
-  const cid = await uploadTextFileToIpfs(appDataDoc)
-  await pinByHash(cid)
-  const { digest } = multihashes.decode(new CID(cid).multihash)
-  return `0x${Buffer.from(digest).toString('hex')}`
+  const { IpfsHash } = await pinJSONToIPFS(appDataDoc)
```

Comment:
You still need to decode the CID to get the multihash digest

The appDataHash must be a hex, but the CID is being stored
![screenshot_2021-11-16_13-06-48](https://user-images.githubusercontent.com/43217/142065818-2a66ac99-9961-4aeb-b413-663eb56b3541.png)

Despite the name, `IpfsHash` is a CID, it seems.